File: Check for BST.java
class Solution {
    // Function to check whether a Binary Tree is BST or not.
    private int MIN = 0;
    private int MAX = 100001;
    
    boolean isbst(Node root, int min, int max) {
        if (root == null)
            return true;
        if (root.data <= min || root.data >= max)
            return false;
            
        return isbst(root.left, min, root.data) 
            && isbst(root.right, root.data, max);
    }
    
    boolean isBST(Node root) {
        // code here.
        return isbst(root, MIN, MAX);
    }
}

File: Check for BST.py
class Solution:
    
    #Function to check whether a Binary Tree is BST or not.
    def isBST(self, root):
        #code here
        self.prev = float('-inf')
        def fun(root):
            if root == None:
                return True
            if fun(root.left) == False:
                return False
            if root.data <= self.prev:
                return False
            self.prev = root.data
            return fun(root.right)
        
        return fun(root)
